File: Academy.Backend/src/CourseManagement/Academy.CourseManagement.Application/DTOs/LessonDto.cs
using Academy.SharedKernel.ValueObjects;

namespace Academy.CourseManagement.Application.DTOs
{
    public class LessonDto
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public int Position { get; set; }
        public string? Content { get; set; } = string.Empty;
        public string LessonType { get; set; } = string.Empty;
        public IReadOnlyList<Question> Questions { get; private set; } = new List<Question>();
        public IReadOnlyList<Attachment> Attachments { get; private set; } = new List<Attachment>();
        public PracticeLessonData? PracticeLessonData { get; private set; }
        public LessonInfoDto? LessonInfoDto { get; private set; }
    }
}

File: Academy.Frontend/src/modules/courses/models/lessonDto.ts
export interface LessonDto {
  id: string;
  title: string;
  position: number;
  content: string;
  lessonType: "TEST" | "LECTURE" | "PRACTICE";
  questions: Question[];
  attachments: Attachment[];
  practiceLessonData: PracticeLessonData;
}

export interface Question {
  title: string;
  answers: Answer[];
}

export interface Answer {
  text: string;
  isCorrect: boolean;
}

export interface Attachment {
  fileName: string;
  fileUrl: string;
}

export interface PracticeLessonData {
  templateCode: string;
  tests: Test[];
}

export interface Test {
  input: string;
  output: string;
}
